src/overlays/bankid/index.tsx

```diff
@@ -0,0 +1,280 @@
+import React from "react";
+
+import { IEcomContext, IEcomStore } from "../../types";
+import { isMobile } from "../../utils/device";
+
+import BankIdOverlay from "./base";
+import StoreAction from "../../constants/store-action";
+
+interface IBankIdProps extends IEcomContext, IEcomStore {
+    onHideOverlay: () => void;
+    onProceedToNextStep: () => void;
+}
+
+interface IState {
+    useQrCode: boolean;
+    cancellationToken?: NodeJS.Timeout;
+}
+
+class BankId extends React.Component<IBankIdProps, IState> {
+    constructor(props: IBankIdProps) {
+        super(props);
+
+        const useQrCode = !isMobile();
+        this.state = {
+            useQrCode,
+            cancellationToken: null,
+        };
+
+        this.cancel = this.cancel.bind(this);
+        this.switchMethod = this.switchMethod.bind(this);
+        this.launch = this.launch.bind(this);
+        this.onCancelled = this.onCancelled.bind(this);
+        this.getSwitchMessage = this.getSwitchMessage.bind(this);
+        this.onCancelledForSwitch = this.onCancelledForSwitch.bind(this);
+    }
+
+    componentDidMount() {
```

Comment:
We really should see if we can remove these legacy methods for lifecycle management! I know we already have a couple of them in the code base, but preferably we shouldn't add any more :-)

Reply:
Is `componentDidMount` considered as legacy?

Reply:
Ah, no - it's not! My intention was to add this comment to the `componentDidUpdate`, but that lifecycle method is not deprecated either. My bad. :-)